yarn-project/simulator/src/common/db_interfaces.ts

```diff
@@ -60,30 +60,23 @@ export interface CommitmentsDB {
 
   /**
    * @param leafIndex the leaf to look up
-   * @returns The l1 to l2 leaf value or undefined if not found.
+   * @returns The l1 to l2 leaf message hash or undefined if not found.
    */
-  getL1ToL2LeafValue(leafIndex: bigint): Promise<Fr | undefined>;
+  getL1ToL2MessageHash(leafIndex: bigint): Promise<Fr | undefined>;
 
   /**
-   * Gets the index of a commitment in the note hash tree.
-   * @param commitment - The commitment.
-   * @returns - The index of the commitment. Undefined if it does not exist in the tree.
-   */
-  getCommitmentIndex(commitment: Fr): Promise<bigint | undefined>;
-
-  /**
-   * Gets commitment in the note hash tree given a leaf index.
+   * Gets note hash in the note hash tree at the given leaf index.
    * @param leafIndex - the leaf to look up.
-   * @returns - The commitment at that index. Undefined if leaf index is not found.
+   * @returns - The note hash at that index. Undefined if leaf index is not found.
    */
-  getCommitmentValue(leafIndex: bigint): Promise<Fr | undefined>;
+  getNoteHash(leafIndex: bigint): Promise<Fr | undefined>;
 
   /**
-   * Gets the index of a nullifier in the nullifier tree.
+   * Checks if a nullifier exists in the nullifier tree.
    * @param nullifier - The nullifier.
-   * @returns - The index of the nullifier. Undefined if it does not exist in the tree.
+   * @returns - True if the nullifier exists in the tree, false otherwise.
    */
-  getNullifierIndex(nullifier: Fr): Promise<bigint | undefined>;
+  checkNullifierExists(nullifier: Fr): Promise<boolean>;
```

Comment:
The only reason why we ever obtained the index was to check if the nullifier exists. Hence made sense to change the API here.